### DISCUSSION TITLE
how to return the column names when querying parquet files

### DISCUSSION THREAD

**Discussion by qiangbo** [duckdb/duckdb#3488] [General]
Hi,

When I use the DuckDB to query multiple parquets, the resulting dataframe always has 0, 1, 2,... as their column names. How to get the column names together with the query?

Below is my python pseudo code.

import pandas as pd
import duckdb

qry = 
"""
select
    date,
    sum(sales) as sales
from
    'C:/data/data_*.parquet'
group by
    date
"""

data = duckdb.query(qry).fetchall()
df_data = pd.DataFrame(data)

Current output data frame:

0                    1 
2022-03-01    100
2022-03-02    200
2022-03-03    300

How to get this?

date                sales
2022-03-01    100
2022-03-02    200
2022-03-03    300

**Comment by Alex-Monahan** [comment]
Hello!

You can actually go directly from DuckDB to Pandas! It is faster that way too.
```python
df_data = duckdb.query(qry).df()
```

**Comment by qiangbo** [comment]
Wow, that is fantastic! Thanks a lot!

**Comment by qiangbo** [comment]
Actually, one of the queries generated the following error message:

Run time Error: Unsupported type LITST<VARCHAR> for DuckDB -> NumPy conversion

One of the column in the parquet file is list of strings.... understand this is a limitation for the conversion, is there a way to get around of that? The following code works fine except the headers are missing.

`data = duckdb.query(qry).fetchall()`
`df_data = pd.DataFrame(data)`

**Comment by Alex-Monahan** [comment]
Which version of DuckDB are you using? Lists are supported in the Python client, so I would have expected this to work!

https://github.com/duckdb/duckdb/pull/2753

**Comment by qiangbo** [comment]
I am using version 0.3.1.

Because I cannot upgrade... is there a manual workaround?

**Reply by Alex-Monahan** [reply]
Hmm, if you can't upgrade, one workaround would be to cast the column as a varchar, export it to Pandas, then reparse it to a list in Python
